gremlin-javascript/pom.xml

```diff
@@ -236,6 +236,16 @@ file.write(file.getText("UTF-8").replaceFirst(/"version": "(.*)",/, "\"version\"
                             <failOnError>true</failOnError>
                         </configuration>
                     </execution>
+                    <execution>
+                      <id>npm lint</id>
+                      <phase>lint</phase>
```

Comment:
I don't remember much about maven but I think this needs to be `<phase>integration-test</phase>` to be able to run.

[I don't see it on the build](https://github.com/apache/tinkerpop/runs/5646493848?check_suite_focus=true).

@spmallette can you help us with maven foo?

Reply:
I missed the arguments section when I copied the block over, I pushed up a new commit and I think I got it

Reply:
that worked but it turns out eslint dropped support for v10 node after version 7. I downgraded locally and everything seems fine, a new check should be running soon

Reply:
> you would want to probably tie this to the "verify" phase which would happen after integration tests

Wouldn't it make more sense to let a linter run before executing costly integration tests? If my code isn't formatted correctly, I wouldn't want to wait after all integration tests were executed to get that information, only so I can change the formatting and then having to execute all integration tests again to see whether the linter is now happy.
How about tying this to the `validate` phase instead which happens first and [the docs](https://maven.apache.org/guides/introduction/introduction-to-the-lifecycle.html#a-build-lifecycle-is-made-up-of-phases) describe it as:

> validate the project is correct and all necessary information is available

Reply:
yeah, I ideally an earlier phase would be great.

I don't have a strong opinion tho, I don't want to drag this PR further and I would like to merge @tkolanko in the next few days.

Reply:
the trouble i had with `validate` was that it was too early if we wanted lint to consider generated sources. if we don't care about that then it's fine to change it. doesn't really bother me either way.